### ISSUE TITLE
Unable to create a basic electron + angular app

### ISSUE THREAD

**Issue by ziacik** [nstudio/xplat#223]
I'm sorry if my issue comes out of my ignorance. I'm a bit frustrated trying to generate an electron + angular workspace. I'm trying to use the steps pointed out at https://nstudio.io/xplat/getting-started/getting-started but it seems I'm missing something.

So I generate a nrwl workspace with "Empty" and "Nx cli". Then I add xplat using npm. Then I try to generate an app with `nx generate app`. The first thing that surprises me is that there is no "angular" option in the wizard's selection. So I try the "electron" option. Oh, now it offers angular to me, my bad. I fill everything up, and bam! 
Error: `Collection "@nstudio/web-angular" cannot be resolved.`

Can't google anything, so I just take a chance and just do `npm install -D @nstudio/web-angular`. It works. Start the generation again, fill everything up, bam! Another error:
`The target app name "web" does not appear to be in workspace.json. You may need to generate the app first or perhaps check the spelling.` Well, but I hadn't had a chance to generate it first.

(Also, why does the wizard ask me "In which directory should the app be generated?" I don't know what to put there.)

Please see the attached video to see my steps better.

![xplat](https://user-images.githubusercontent.com/4528506/98450022-4c300500-2139-11eb-8a4e-e429ddf4ead4.gif)

**Comment by ziacik** [comment]
Update: I managed to generate the angular app first using `nx g @nrwl/angular:app`. Then the `nx generate app` was able to finish. 

But, then `npm run start.electron.app` still didn't work. Why? Some `xplat` stuff wasn't generated. So I figured somehow that running `nx g @nstudio/xplat:init` would generate stuff (`libs/core` and such).

Now trying to run the electron app again, I get this:
```
A JavaScript error occurred in the main process
Uncaught Exception:
/home/ziacik/Project/teee/apps/electron-app/src/index.js:1
import { app, BrowserWindow, ipcMain, screen } from 'electron';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at Module._compile (internal/modules/cjs/loader.js:815:22)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:892:10)
    at Module.load (internal/modules/cjs/loader.js:735:32)
    at Module._load (internal/modules/cjs/loader.js:648:12)
    at Module._load (electron/js2c/asar.js:717:26)
    at Function.Module._load (electron/js2c/asar.js:717:26)
    at loadApplicationPackage (/home/ziacik/Project/teee/node_modules/electron/dist/resources/default_app.asar/main.js:109:16)
    at Object.<anonymous> (/home/ziacik/Project/teee/node_modules/electron/dist/resources/default_app.asar/main.js:155:9)
    at Module._compile (internal/modules/cjs/loader.js:880:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:892:10)
```

Oh, and I also had to create a `tsconfig.electron.json` file in the angular app that would include `main.electron.ts` and change the `workspace.json` file to point to that one instead of `tsconfig.app.json`.

**Comment by sr3dna** [comment]
@ziacik , I encountered the same issue.   I was able to get it to work after specifying a compilerOption.   In the tsconfig.json file under my electron-app folder, 

{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
        "types": ["node", "jest"],
         **"module": "CommonJS",    // Added this line**
    },
  "include": ["src/index.ts"]
}

Hope this works for you too.

**Comment by NathanWalker** [comment]
Resolved in 11.0.21 - to create an Electron app keep in mind you'll need to have a web app generated already to use as the electron target, ie (replace `sample` with name of app you'd like to create):

```
nx g app sample
```
> choose `web`

```
nx g app desktop --target=web-sample
```
> choose `electron`

There will now be an Electron app ready to develop which wraps the `web-sample` app with live reload enabled by running:

```
npm run start.electron.desktop
```

**Comment by peb7268** [comment]
Thanks for this work and this answer @NathanWalker ! 🙌
